components/CreatePost.tsx

```diff
@@ -0,0 +1,102 @@
+import React, { useState } from "react";
+
+type NewPost = {
+  titleDescription: string;
+  type: string;
+};
+
+const iniState = {
+  titleDescription: "",
+  type: "Life",
+};
+
+export const CreatePost = () => {
+  const POST_TYPES = ["Life", "Job", "Joke", "Animals", "Food"];
+  const [isHidden, setIsHidden] = useState(true);
+  const [newPost, setNewPost] = useState<NewPost>(iniState);
+
+  const isEmpty = (str: string) => {
+    const length = str.trim().length;
+    return length < 1;
+  };
```

Comment:
what's this for? this logic can be simplified into just
`!str.trim()`

Reply:
so we really don't need this helper function especially on our use cases in this code